pkg/webhook/parser.go

```diff
@@ -141,7 +146,61 @@ func (c Config) ToWebhook() admissionreg.Webhook {
 			// Put "\n" as an placeholder as a workaround til 1.13+ is almost everywhere.
 			CABundle: []byte("\n"),
 		},
+	}, nil
+}
+
+// ToValidatingWebhook converts this rule to its Kubernetes API form.
+func (c Config) ToValidatingWebhook() (admissionreg.ValidatingWebhook, error) {
+	if c.Mutating {
+		return admissionreg.ValidatingWebhook{}, fmt.Errorf("%s is a mutating webhook", c.Name)
+	}
+
+	whConfig := admissionreg.RuleWithOperations{
+		Rule: admissionreg.Rule{
+			APIGroups:   c.Groups,
+			APIVersions: c.Versions,
+			Resources:   c.Resources,
+		},
+		Operations: make([]admissionreg.OperationType, len(c.Verbs)),
 	}
+
+	for i, verbRaw := range c.Verbs {
+		whConfig.Operations[i] = verbToAPIVariant(verbRaw)
+	}
+
+	// fix the group names, since letting people type "core" is nice
+	for i, group := range whConfig.APIGroups {
+		if group == "core" {
+			whConfig.APIGroups[i] = ""
+		}
+	}
+
+	var failurePolicy admissionreg.FailurePolicyType
+	switch strings.ToLower(c.FailurePolicy) {
+	case strings.ToLower(string(admissionreg.Ignore)):
+		failurePolicy = admissionreg.Ignore
+	case strings.ToLower(string(admissionreg.Fail)):
+		failurePolicy = admissionreg.Fail
+	default:
+		failurePolicy = admissionreg.FailurePolicyType(c.FailurePolicy)
+	}
+	path := c.Path
+	return admissionreg.ValidatingWebhook{
+		Name:          c.Name,
+		Rules:         []admissionreg.RuleWithOperations{whConfig},
+		FailurePolicy: &failurePolicy,
+		ClientConfig: admissionreg.WebhookClientConfig{
```

Comment:
can we refactor the more complicated bits out into methods, so it'll be harder to accidentally change on but not the other?

Reply:
+1
Any update?

Reply:
@DirectXMan12 #318 is a refactor that is hopefully in keeping with you request.

Reply:
cool, thanks!